doc/JSON-RPC-interface.md

```diff
@@ -5,6 +5,36 @@ The headless daemon `bitcoind` has the JSON-RPC API enabled by default, the GUI
 option. In the GUI it is possible to execute RPC methods in the Debug Console
 Dialog.
 
+## Endpoints
+
+There are two JSON-RPC endpoints on the server:
+
+1. `/`
+2. `/wallet/<walletname>/`
+
+### `/` endpoint
+
+This endpoint is always active and can service both wallet and non-wallet
+requests when there are zero or one wallets loaded.
+
+### `/wallet/<walletname>/` endpoint
+
+This endpoint is only activated when the wallet component has been compiled in
+and can service both wallet and non-wallet requests. It MUST be used for wallet
+requests when two or more wallets are loaded.
+
+This endpoint is equivalent to using bitcoin-cli with the `-rpcwallet=` parameter.
+
+Best practice would dictate using the `/wallet/<walletname>/` endpoint for ALL
+requests when multiple wallets are in use.
+
+### Quirks:
+
+* With more than one wallet loaded the `/` endpoint will still be able to
+service non-wallet requests.
+* When active the `/wallet/<walletname>/` endpoint can still be used when no
+wallet is loaded by providing a dummy `walletname`.
```

Comment:
I think these can be folded in above if you like my suggestions.

I think a few `curl` examples might be appropriate here as well since this document is covering raw endpoint stuff

Reply:
Again I think I agree here, especially seeing as there are no /wallet/ examples given in RPCHelp

Reply:
Removed quirks